packages/hint-compat-api/src/utils/browsers.ts

```diff
@@ -0,0 +1,59 @@
+import * as semver from 'semver';
+
+const coerce = (version: string): semver.SemVer | string => {
+    return semver.coerce(version) || /* istanbul ignore next */ version;
+};
+
+/**
+ * Serialize condensed version ranges for provided browsers.
+ *
+ * ```js
+ * joinBrowsers(['chrome 74', 'chrome 75', 'chrome 76', 'edge 15', 'edge 16', 'firefox 67']);
+ * // returns 'chrome 74-76, edge 15-16, firefox 67';
+ * ```
+ */
+export const joinBrowsers = (browsers: string[]): string => {
+    const versionsByName = new Map<string, string[]>();
+
+    // Group browser versions by browser name.
+    for (const browser of browsers) {
+        const [name, version] = browser.split(' ');
+
+        if (!versionsByName.has(name)) {
+            versionsByName.set(name, []);
+        }
+
+        versionsByName.get(name)!.push(version);
+    }
+
+    const results: string[] = [];
+
+    // Sort and serialize version ranges for each browser name.
+    for (const [name, versions] of versionsByName) {
+        versions.sort((v1, v2) => {
+            return semver.compare(coerce(v1), coerce(v2));
+        });
+
+        const ranges = [];
+
+        // Outer loop runs once per start of each range.
+        for (let i = 0; i < versions.length; i++) {
+            const start = versions[i];
+
+            // Inner loop jumps to the end of a range.
+            while (parseInt(versions[i + 1]) - parseInt(versions[i]) <= 1) {
+                i++;
```

Comment:
@sarvaje You wanted to think about a different approach to doing this without modifying `i` inside the loop.

Reply:
@sarvaje Here's what I ended up changing it to:

```ts
for (let i = 0, start = versions[0]; i < versions.length; i++) {
    if (parseInt(versions[i + 1]) - parseInt(versions[i]) <= 1) {
        continue; // Continue until the end of a range.
    }

    // Format current range as either `start` or `start-end`.
    if (start === versions[i]) {
        ranges.push(start);
    } else {
        ranges.push(`${start}-${versions[i]}`);
    }

    // Remember the start of the next range.
    start = versions[i + 1];
}
```

Reply:
I like this version more than the previous one hehehe.